src/lib/utils.ts

```diff
@@ -1,7 +1,29 @@
+import { formatDistanceToNow, type FormatDistanceToNowOptions } from 'date-fns';
+
 export function generateStorageId() {
 	return Math.random().toString(36).substring(2, 8); // E.g. `z7avx9`
 }
 
 export function getUpdatedAtDate() {
 	return new Date().toISOString();
 }
+
+export function formatShortDistanceToNow(
+	date: Date | number | string,
+	options?: FormatDistanceToNowOptions
+) {
+	const longFormat = formatDistanceToNow(date, options);
+	const shortFormat = longFormat
+		.replace(' seconds ago', 's')
+		.replace(' minute ago', 'm')
+		.replace(' minutes ago', 'm')
+		.replace(' hour ago', 'h')
+		.replace(' hours ago', 'h')
+		.replace(' day ago', 'd')
+		.replace(' days ago', 'd')
+		.replace(' month ago', 'mo')
+		.replace(' months ago', 'mo')
+		.replace(' year ago', 'y')
+		.replace(' years ago', 'y');
+	return shortFormat;
+}
```

Comment:
I think it'd be easier to use your original library suggestion.
The are some edge-cases not covered by this function, for example:

![image](https://github.com/user-attachments/assets/1eeb1398-27d8-407a-aade-8bc2dbb9d32b)

I usually don't dislike the idea of writing our own _"simple helper functions"_ (if we can avoid using a 3rd-party library) but in this case I think it's likely there are other edge-cases that we haven't seen yet, so it's best to stick to a solution that's more mature.

Thankfully, we don't really need to do complex date manipulation so we can probably skip on `date-fns` for now.

Also, eventually I'd like to localize these strings but we'll cross that bridge when we have to.

Reply:
I agree!

I have tried installing `dayjs-twitter` but I couldn't make it work, I think it's not made to work with typescript. I have searched for other libraries but I didn't find any.

In the mean time, I reverted it to the `date-fns` format which is too long but at least we have the edge cases covered.

Do you have any other ideas? At this point I don't think it would be a bad idea to write a custom function which covers all the edge cases, unless we find another library that does that for us.

Reply:
No problem, `date-fns` with the default `formatDistanceToNow()` will work 👍